lib/gds_api/publishing_api_v2.rb

```diff
@@ -53,6 +53,42 @@ def lookup_content_ids(base_paths:)
     response.to_hash
   end
 
+  # Get the current draft and/or the current live edition matching a set of base paths.
+  # "draft" and "live" refer to the content store the content is in.
+  # The live edition contains an `unpublishing` hash if the edition is unpublished.
+  # An unpublishing hash has `type` and `redirect` keys matching the unpublish request.
+  #
+  # The request accepts both GET/POST. We use POST here to avoid long URIs.
+  # @param base_paths [Array]
+  # @return [Hash] a hash keyed by `base_path`. The value is nil if the base path is not used or belongs to an access limited draft.
+  # @example
+  #
+  #   publishing_api.lookup_by_base_paths(['/pay-vat', '/nothing'])
+  #   # => {"/pay-vat"=>{"live"=>{"content_id"=>"a484eaea-eeb6-48fa-92a7-b67c6cd414f6", "locale"=>"en", "document_type"=>"guide"}}, "/nothing"=>nil}
+  #
+  # @see https://github.com/alphagov/publishing-api/blob/master/doc/api.md#get-v2lookup-by-base-path
+  def lookup_by_base_paths(base_paths)
+    response = post_json("#{endpoint}/v2/lookup-by-base-path", base_paths: base_paths)
```

Comment:
any chance we could check the length of the string to determine whether it is get or post? It'd be preferable if it was mostly a get rather than post